### DISCUSSION TITLE
Upgrade installation failure

### DISCUSSION THREAD

**Discussion by chickman-copa** [jamessimone/apex-rollup#327] [Q&A] [answered]
Hello,

Trying to go from 1.4.5 > 1.5.7 and getting the following errors:

Your request to install package "Apex Rollup Updating documentation, fixed issue with hierarchical rollups not always querying all" was unsuccessful. None of the data or setup information in your [salesforce.com](http://salesforce.com/) organization was affected.

If your install continues to fail, contact Salesforce CRM Support through your normal channels and provide the following information.

Organization:
User: Christopher Hickman
Package: Apex Rollup (04t6g000008SkNk)

Problem:

1.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 18, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

2.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 29, column 43: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

3.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 33, column 14: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

4.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 42, column 43: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

5.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 52, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

6.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 63, column 49: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

7.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 73, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

8.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 112, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

9.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 124, column 14: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

10.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 147, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

11.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 162, column 14: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

12.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 184, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

13.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 206, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

14.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 216, column 46: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

15.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 217, column 46: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

16.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 233, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

17.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 256, column 50: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

18.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 257, column 41: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

19.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 259, column 48: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

20.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 287, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

21.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 302, column 48: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

22.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 303, column 48: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

23.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 316, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

24.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 326, column 48: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

25.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 333, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

26.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 353, column 44: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

27.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 368, column 48: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

**Comment by jamessimone** [comment, marked as answer]
@chickman-copa hey chris, good to hear from you! Unfortunately, you've encountered the breaking change between 1.4 and 1.5 - I didn't catch this while releasing 1.4.11 and made the change to 1.5.x shortly thereafter. You should be able to choose the advanced options [for 1.4.10](https://test.salesforce.com/packaging/installPackage.apexp?p0=04t6g000008SjuzAAC) when installing Apex Rollup, and use the "Deprecate Only" upgrade mode - doing that _should_ allow you to upgrade smoothly to one of the 1.5.x versions thereafter.

I try - as much as is possible - to prevent breaking changes between package versions, so I apologize I've stuck with you a non-standard upgrade route. Hopefully the deprecate only method will work in order to get you back on track. Please let me know!

**Reply by jamessimone** [reply]
Also, realizing some of those failures are from tests - not sure if you had the `Extra Tests` plugin installed or if you had deployed the tests manually, but you may have an even easier upgrade path by simply deleting the `RollupRelationshipFieldFinderTests` and re-installing the extra tests plugin/re-deploying the metadata for that specific test.

**Reply by chickman-copa** [reply]
Thanks! So, the link above for 1.4.10, is actually for a 0.0.8 version of the Extra Tests package. But based on the above, I uninstalled the 0.0.7 version I had, tried the latest main package, and it worked! I then tried the 0.0.8 Extra Tests just to see, and got this:

Your request to install package "Apex Rollup - Extra Code Coverage Updating code coverage" was unsuccessful. None of the data or setup information in your [salesforce.com](http://salesforce.com/) organization was affected.

If your install continues to fail, contact Salesforce CRM Support through your normal channels and provide the following information.

Organization: 
User: Christopher Hickman
Package: Apex Rollup - Extra Code Coverage (04t6g000008Sjuz)
Error Number: 1705132668-370690 (-1841584367)
Problem:

1.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

2.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

3.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

4.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

5.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

6.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

7.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

8.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

9.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

10.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

11.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

12.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

13.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

14.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

15.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

16.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

17.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

18.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

19.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

20.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

21.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

22.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

23.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

24.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

25.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

26.  Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)
RollupRelationshipFieldFinderTests: Constructor not defined: [RollupRelationshipFieldFinder].<Constructor>(RollupControl__mdt, Rollup__mdt, Set<String>, Schema.SObjectType, Map<Id,SObject>)

27.  Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal
RollupRelationshipFieldFinderTests: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

**Reply by jamessimone** [reply]
@chickman-copa that version of the extra code coverage plugin is also out of date! Where were you getting the link for that? The [0.0.11](https://test.salesforce.com/packaging/installPackage.apexp?p0=04t6g000008SkNaAAK) version of the plugin should be the one currently listed on the README, and that will work for you no problem.

Apologies for the difficulties! Hopefully this all gets worked out soon.

**Comment by chickman-copa** [comment]
Clicking on "for 1.4.10" in your 1st response takes you to package ID 04t6g000008SjuzAAC, which is 0.0.8 of extra code coverage (probably just a copy/paste error). When I click on "0.0.11" above, it takes me to the right install link, and I was able to successfully install. Thanks!

**Reply by jamessimone** [reply]
Whoops, sorry about that - I was copying and pasting to search in the code base to ensure I had properly updated the Readme link - apologies for the mixup but I’m glad it’s all sorted. Again, always good hearing from you!
